### PR TITLE
Fix tagging for subscriptions

### PR DESCRIPTION
**Pull request by MarcelHeek**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/ISSUE-ID-GOES-HERE)

## PR Checklist

---

- [ ] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [ ] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Does this introduce a breaking change

- [ ] YES
- [ ] NO

### CODE REVIEW

File: modules/subscriptions/subscriptions.tf [hunk 1]
```diff
@@ -21,7 +21,11 @@ resource "azurerm_subscription" "sub" {
   subscription_id   = try(var.settings.subscription_id, null) != null ? var.settings.subscription_id : null
   billing_scope_id  = try(var.settings.billing_scope_id, null) == null ? try(data.azurerm_billing_enrollment_account_scope.sub.0.id, data.azurerm_billing_mca_account_scope.sub.0.id, null) : var.settings.billing_scope_id
   workload          = try(var.settings.workload, null)
-  tags              = try(var.settings.tags, null)
+  #tags              = try(var.settings.tags, null)
+  tags = merge(
+    var.tags,
+    lookup(var.settings, "tags", {})
+  )
```
Comment (JvHd-vw) [inline modules/subscriptions/subscriptions.tf:26]: onderstaande merge in subscriptions.tf

merge(lookup(each.value, "tags", {}), var.tags) 

is hetzelfde als die in modules/subscriptions/subscriptions.tf
tags = merge(
  var.tags,
  lookup(var.settings, "tags", {})
)

Comment (JvHd-vw) [reply modules/subscriptions/subscriptions.tf:26]: tags              = try(var.tags, null)  zou ik hier verwachten

Comment (MarcelHeek) [reply modules/subscriptions/subscriptions.tf:26]: @JvHd-vw Doe ik wat je voorsteld, dan werkt de overload van tags binnen de subscriptions.tfvars niet. Dus dat levert niet het gewenste resultaat op.

Comment (MarcelHeek) [reply modules/subscriptions/subscriptions.tf:26]: @JvHd-vw goede tip, thanks

**Review by JvHd-vw — APPROVED** [review]

**Pull request merged**